dbt/adapters/databricks/api_client.py

```diff
@@ -300,7 +302,13 @@ def poll_for_completion(self, command: CommandExecution) -> None:
             terminal_states={"Finished", "Error", "Cancelled"},
             expected_end_state="Finished",
             unexpected_end_state_func=self._get_exception,
-        )
+        ).json()
+
+        if response["results"]["resultType"] == "error":
```

Comment:
Lost this in the refactor: for some reason Command exec will give a state of 'Finished' rather then 'Error' some times, and then stuff the error in the results.

Reply:
what if the `result` or `resultType` does not exist

Reply:
these are published parts of the Databricks REST API.  If these don't exist, the entire API becomes untrustworthy.